tests/component/system/test_system.py

```diff
@@ -11,3 +16,40 @@ def test___get_analog_power_up_states_with_output_type___returns_power_up_states
         assert channel_names[i] == power_up_states[i].physical_channel
         assert 0.0 == power_up_states[i].power_up_state
         assert PowerUpChannelType.CHANNEL_HIGH_IMPEDANCE == power_up_states[i].channel_type
+
+
+def test_valid_power_up_states___set_analog_power_up_states_with_output_type___sets_power_up_states_without_errors(
+    system,
+):
+    channel_names = ["aoTester/ao0", "aoTester/ao1"]
+    power_up_states = [
+        AOPowerUpState(
+            physical_channel=channel_name,
+            power_up_state=-1.0,
+            channel_type=PowerUpChannelType.CHANNEL_HIGH_IMPEDANCE,
+        )
+        for channel_name in channel_names
+    ]
+
+    result = system.set_analog_power_up_states_with_output_type(power_up_states)
+
+    assert result == None
+
+
+def test_invalid_power_up_states___set_analog_power_up_states_with_output_type___throws_invalid_attribute_value_error(
+    system,
+):
+    channel_names = ["aoTester/ao0", "aoTester/ao1"]
+    power_up_states = [
+        AOPowerUpState(
+            physical_channel=channel_name,
+            power_up_state=20,
+            channel_type=PowerUpChannelType.CHANNEL_VOLTAGE,
+        )
+        for channel_name in channel_names
+    ]
```

Comment:
Can you modify or add a new test case where only the second power up state is bad? That way we know that its actually iterating through all of them.